transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/MessagingClient.kt

```diff
@@ -10,15 +10,48 @@ import com.genesys.cloud.messenger.transport.util.ErrorCode
 interface MessagingClient {
 
     /**
-     * MessagingClient state.
+     * Container that holds all possible MessagingClient states.
      */
     sealed class State {
+        /**
+         * MessagingClient was instantiated, but never attempted to connect.
+         */
         object Idle : State()
+
+        /**
+         * Trying to establish secure connection via WebSocket.
+         */
         object Connecting : State()
+
+        /**
+         * Secure connection with WebSocket was opened.
+         */
         object Connected : State()
+
+        /**
+         * Session was successfully configured.
+         *
+         * @property connected true if configuration were established and session open.
+         * @property newSession indicates if configured session is new. When configuring an existing session [newSession] will be false.
+         */
         data class Configured(val connected: Boolean, val newSession: Boolean?) : State()
+
+        /**
+         * Invoked when the remote peer has indicated that no more incoming messages will be transmitted.
```

Comment:
Instead of `Invoked when …` which makes it seem a bit like this is a listener, how about `State when …`? (same comment applies to the `Closed` state below.